app/UpdateStore.tsx

```diff
@@ -75,8 +81,10 @@ export function UpdateStore(props: {
             if (suburb === "") {
               return setErrorMessage("Suburb cannot be empty");
             }
+            setAwaitingUpdateStoreCheck(true);
             await updateStore(store, displayName, fullName, suburb);
             onUpdateStoreSuccess();
+            setAwaitingUpdateStoreCheck(false);
```

Comment:
here you do actually wait for the `updateStore` method to finish before re-enabling the button. I also just realised that you don't do the same checks as the creation flow though, what happens if you try to update a store to have the same display name + suburb as an existing one?

Reply:
the reason it doesn't do the same checks right now is because if they were in place, it wouldn't allow you to update at all because it would say store already exists if the display name + suburb OR full name + suburb combo already exists. 

if you try to update  store that has the same display name + suburb as an existing one, prisma will throw an error because we asked displayname to be a unique field

Reply:
we could add a condtion where if the store ID is the same as the one we're editing, it ignores if all the details are the same to avoid throwing errors?

Reply:
yep that's the correct thing to do, look for any clashes that are not the one we are editing, which is one of the reasons to have a unique ID that is not bound to any data about the entity (eg we don't make `username` the primary key of users`